src/main/java/org/spongepowered/mod/mixin/core/fml/common/crafting/MixinCraftingHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.mixin.core.fml.common.crafting;
+
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.fml.common.ModContainer;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.io.File;
+
+@Mixin(CraftingHelper.class)
+public class MixinCraftingHelper {
+
+    @Redirect(method = "findFiles(Lnet/minecraftforge/fml/common/ModContainer;Ljava/lang/String;Ljava/util/function/Function;"
+            + "Ljava/util/function/BiFunction;ZZ)Z", at = @At(value = "INVOKE", ordinal = 0,
+            target = "Lnet/minecraftforge/fml/common/ModContainer;getSource()Ljava/io/File;"))
+    private static File onFindFiles(ModContainer mod) {
+        final File source = mod.getSource();
+        if (!source.isDirectory() || new File(source, "assets").exists()) {
+            return source;
+        }
+        // TODO: Clean way to handle this? Fix this in forge?
+        // In a development environment are the resources and classes separated
+        // into two source directories, at least in IntelliJ.
+        // This is a temporary fix to make sure that advancement/recipe files
+        // are loaded in the IDE.
```

Comment:
I would love some input on this ugly fix, not idea how to work around it.

Reply:
Is this for SpongeForge itself or for Sponge plugins loaded from the classpath?

Reply:
Sponge plugins loaded from the classpath.

Reply:
Generally, we set `inheritOutputDirs = true` for all IntelliJ IDEA modules, which makes IntelliJ IDEA write classes and resources into one shared output directory. This is applied automatically for SpongeAPI and the Sponge implementation modules and all plugins using SpongeGradle.

However, I noticed that it wasn't applied to `testplugins` which might be the reason for the problem if that's what you were testing with. I fixed that in https://github.com/SpongePowered/SpongeAPI/commit/25890783ac54848846bfe21e1542dfbf5b8e38e2, can you check if your workaround is still necessary after this commit?

If you are using a separate plugin for testing, make sure you apply the SpongeGradle plugin.

Reply:
That worked, thanks! :)